src/main/java/seedu/tinner/storage/JsonAdaptedRole.java

```diff
@@ -80,16 +84,14 @@ public Role toModelType() throws IllegalValueException {
         }
         final Status roleStatus = new Status(status);
 
-        if (deadline == null) {
+        if (reminderDate == null) {
```

Comment:
Based on our previous implementation, we removed null checks for optional fields as it is possible for these fields to be null. Should this apply to reminder date as well?

Reply:
Can someone weigh in on this? @petermonky @BrendonLau @jinnhl? I believe that with our current implementation none of those optional fields will be null. They are empty strings but not null values. Correct me if I'm wrong.

Reply:
> I believe that with our current implementation none of those optional fields will be null.

Does this apply to other classes as well? If so then I think we can add `null`checks the other classes too, just to be more defensive.

Reply:
Yes I have updated null checks for them. Testing has yet to be updated.